python/ray/tune/cloud.py

```diff
@@ -0,0 +1,321 @@
+import os
+import shutil
+import subprocess
+import tempfile
+from typing import Optional
+
+from ray import logger
+from ray.tune.sync_client import (S3_PREFIX, GS_PREFIX, HDFS_PREFIX,
+                                  ALLOWED_REMOTE_PREFIXES)
+from ray.util import PublicAPI
+
+
+def is_cloud_target(target: str):
+    return any(target.startswith(prefix) for prefix in ALLOWED_REMOTE_PREFIXES)
+
+
+def _clear_bucket(bucket: str):
+    if not is_cloud_target(bucket):
+        raise ValueError(
+            f"Could not clear bucket contents: "
+            f"Bucket `{bucket}` is not a valid or supported cloud target.")
+
+    try:
+        if bucket.startswith(S3_PREFIX):
```

Comment:
The ifs may get hard to maintain if we ever want to add support for other clouds. Maybe it would be a good idea to have a small class defining commands for a cloud, and then have a list/dict of them.

Reply:
I 2nd this sentiment. ideally we would have all storage providers implement a common api class, so commands and edge cases are grouped per provider.
but I understand that is a much bigger change, and if you'd rather do it in a followup change.

Reply:
We'll definitely do it (and presumably get rid of the command line interface and use proper SDKs instead), but it's out of scope for this PR.